handlers/whatsapp/whatsapp.go

```diff
@@ -147,6 +219,39 @@ func (h *handler) receiveEvent(ctx context.Context, channel courier.Channel, w h
 	return events, courier.WriteDataResponse(ctx, w, http.StatusOK, "Events Handled", data)
 }
 
+func resolveMediaURL(channel courier.Channel, mediaID string) (string, error) {
```

Comment:
This method could alo be responsible to upload the media file and get its id to use in the message

Reply:
I'm not entirely sure what you're suggesting here . 

I'm happy to split out the uploading of media to WA to a different function if you feel that would be better. I do think keeping the the download stuff separate from the upload stuff is a good idea though.